staging/src/k8s.io/apiserver/pkg/server/openapi/openapi.go

```diff
@@ -64,14 +92,36 @@ func RegisterOpenAPIService(servePath string, webServices []*restful.WebService,
 		return err
 	}
 
-	mux.UnlistedHandleFunc(servePath, func(w http.ResponseWriter, r *http.Request) {
-		resp := restful.NewResponse(w)
-		if r.URL.Path != servePath {
-			resp.WriteErrorString(http.StatusNotFound, "Path not found!")
-		}
-		// TODO: we can cache json string and return it here.
-		resp.WriteAsJson(o.swagger)
-	})
+	type fileInfo struct {
+		ext  string
+		data []byte
+		mime string
+	}
+
+	files := []fileInfo{
+		{".json", o.swaggerBytes, MIME_JSON},
+		{".pb", o.swaggerPb, MIME_PB},
+		{".pb.gz", o.swaggerPbGz, MIME_PB_GZ},
```

Comment:
While I love reusing code, adding dependencies to gogep is painful and this is only 3-4 line more code here. I would like to keep it if it is not that bad.